src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -342,7 +344,7 @@ var ReactMultiChild = {
           );
         }
         nextIndex++;
-        lastPlacedNode = ReactReconciler.getNativeNode(nextChild);
+        lastPlacedNode = ReactReconciler.getLastNativeNode(nextChild);
```

Comment:
Okay, it seems to work out nicely if I just make the closing comment node the "native node" and cache the opening comment node on the text component. The opening comment is only needed when calling `removeDelimitedText` or `replaceDelimitedText`. Or I suppose those functions could traverse backwards instead, if they're given the domID. I'll push the changes and let you be the judge. :)